runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/Credentials.kt

```diff
@@ -17,4 +17,5 @@ public data class Credentials(
     val sessionToken: String? = null,
     val expiration: Instant? = null,
     val providerName: String? = null,
+    val nextRefresh: Instant? = null,
```

Comment:
Let's add documentation for all these fields while we are in here making changes (accessKey, secretKey, and sessionToken may be more obvious but this new field in particular and how it interacts with `nextRefresh` warrants some documentation)